modules/openapi-generator/src/main/resources/typescript-axios/apiInner.mustache

```diff
@@ -369,3 +369,40 @@ export class {{classname}} extends BaseAPI {
     {{/operation}}
 }
 {{/operations}}
+{{#hasEnums}}
+
+{{#operations}}
+{{#operation}}
+{{#allParams}}
+{{#isEnum}}
+{{#stringEnums}}
+/**
+  * @export
+  * @enum {string}
+  */
+export enum {{operationIdCamelCase}}{{enumName}} {
+{{#allowableValues}}
+    {{#enumVars}}
+    {{{name}}} = {{{value}}}{{^-last}},{{/-last}}
+    {{/enumVars}}
+{{/allowableValues}}
+}
+{{/stringEnums}}
+{{^stringEnums}}
+/**
+ * @export
+ */
+export const {{operationIdCamelCase}}{{enumName}} = {
+{{#allowableValues}}
+    {{#enumVars}}
+    {{{name}}}: {{{value}}}{{^-last}},{{/-last}}
+    {{/enumVars}}
+{{/allowableValues}}
+} as const;
+export type {{operationIdCamelCase}}{{enumName}} = typeof {{operationIdCamelCase}}{{enumName}}[keyof typeof {{operationIdCamelCase}}{{enumName}}];
+{{/stringEnums}}
```

Comment:
The block the generates different enum types duplicates existing enum template defined in `modelEnum.mustache`. Suggest reusing that - will make it easier to maintain going forward.

The main difference is that  `modelEnum.mustache` uses the `classname` property of the enum, not `enumName`. So in the Java code you might need to update that to match your desired naming convention (prefix with the operation name)

Reply:
`CodegenParameter`s don't have a `classname` property (it's only in `CodegenModel`s), which is the reason I opted for doing it this way (also because it's the way the fetch generator does it too).

Reply:
@amakhrov any suggestion on how to go about this? should i make another template for params, or make it conditional (like use `classname` if exists, else use `enumName` but that seems kind of hacky?)

Reply:
This is unfortunate.
Looks like separate implementation for CodegenParameter, as you have in your PR is the best way forward right now.